### PR TITLE
Adding known issues in Spark documentation

### PR DESCRIPTION
**Pull request by rohitsinha54**
Issues: 

https://issues.cask.co/browse/CDAP-2821

https://issues.cask.co/browse/CDAP-2949

### CODE REVIEW

File: cdap-docs/developers-manual/source/building-blocks/spark-programs.rst [hunk 1]
```diff
@@ -225,3 +225,22 @@ Spark programs in CDAP can also be added to a :ref:`workflow <workflows>`, simil
...
+    Spark 1.3: https://spark.apache.org/docs/1.3.0/mllib-guide.html#dependencies
+    Spark 1.4: https://spark.apache.org/docs/1.4.0/mllib-guide.html#dependencies
+
+- Unable to resolve methods imported implicitly
+
+Your IDE might fail to resolve methods which are imported implicitly while writing Spark programs. reduceByKey() in PairRDDFunctions class is an example of such methods. If you are using Spark 1.2 or 1.3 to write your application you can resolve this issue by adding the following import explicitly::
+
+    import org.apache.spark.SparkContext._
+
+Note: This issue is got resolved in Spark 1.4 and you will not need an explicit import.
```
Comment (anwar6953) [inline cdap-docs/developers-manual/source/building-blocks/spark-programs.rst]: 'is got' -> 'is'

File: cdap-docs/developers-manual/source/building-blocks/spark-programs.rst [hunk 2]
```diff
@@ -225,3 +225,22 @@ Spark programs in CDAP can also be added to a :ref:`workflow <workflows>`, simil
...
+
+Spark's Mlib has various dependencies. Please make sure you have the appropriate dependencies according to your installation::
+
+    Spark 1.2: https://spark.apache.org/docs/1.2.0/mllib-guide.html#dependencies
+    Spark 1.3: https://spark.apache.org/docs/1.3.0/mllib-guide.html#dependencies
+    Spark 1.4: https://spark.apache.org/docs/1.4.0/mllib-guide.html#dependencies
+
+- Unable to resolve methods imported implicitly
+
+Your IDE might fail to resolve methods which are imported implicitly while writing Spark programs. reduceByKey() in PairRDDFunctions class is an example of such methods. If you are using Spark 1.2 or 1.3 to write your application you can resolve this issue by adding the following import explicitly::
```
Comment (anwar6953) [inline cdap-docs/developers-manual/source/building-blocks/spark-programs.rst]: insert a comma after application

Comment (anwar6953) [inline cdap-docs/developers-manual/source/building-blocks/spark-programs.rst]: Also, I think 'such methods' -> 'such a method'

**Comment by sreevatsanraman** [comment]
Review comments are not addressed. Closing the PR. Please reopen against release/3.1 after addressing the comments

**Comment by rohitsinha54** [comment]
Reopening this PR. To get feedback from John as he needs to make a call on where this should be added and how. Will fix all comments and open and against release branch.

**Pull request closed without merging**